svix-cli/src/main.rs

```diff
@@ -0,0 +1,148 @@
+use crate::cmds::api::authentication::AuthenticationArgs;
+use crate::cmds::api::endpoint::EndpointArgs;
+use crate::cmds::api::event_type::EventTypeArgs;
+use crate::cmds::api::integration::IntegrationArgs;
+use crate::cmds::api::message::MessageArgs;
+use crate::cmds::api::message_attempt::MessageAttemptArgs;
+use crate::cmds::listen::ListenArgs;
+use crate::cmds::open::OpenArgs;
+use crate::cmds::signature::SignatureArgs;
+use crate::config::Config;
+use anyhow::Result;
+use clap::{Parser, Subcommand};
+use clap_complete::Shell;
+use cmds::api::application::ApplicationArgs;
+use colored_json::{ColorMode, Output};
+use concolor_clap::{Color, ColorChoice};
+
+mod cli_types;
+mod cmds;
+mod config;
+mod json;
+mod relay;
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+const BIN_NAME: &str = env!("CARGO_BIN_NAME");
+
+#[derive(Parser)]
+#[command(version, about, long_about = None, bin_name = BIN_NAME)]
+#[clap(color = concolor_clap::color_choice())]
+struct Cli {
+    #[command(flatten)]
+    color: Color,
+    #[command(subcommand)]
+    command: RootCommands,
+}
+
+impl Cli {
+    /// Converts the selected `ColorChoice` from the CLI to a `ColorMode` as used by the JSON printer.
+    /// When the color choice is "auto", this considers whether stdout is a tty or not so that
```

Comment:
```suggestion
    /// Converts the selected `ColorChoice` from the CLI to a `ColorMode` as used by the JSON printer.
    ///
    /// When the color choice is "auto", this considers whether stdout is a tty or not so that
```